### ISSUE TITLE
[Ask for help] Bug in testTableSourceBatchDescriptor with Flink 1.10 update

### ISSUE THREAD

**Issue by AlexanderZhao1** [pravega/flink-connectors#341]
**Problem description**
Testcase `testTableSourceUsingDescriptor` contains 2 parts:
testTableSourceStreamingDescriptor and testTableSourceBacthDescriptor
But the batch part can not pass with Flink 1.10 update.

The original way of creating table is:
`tableEnv.registerTableSource("MyTableRow", source);`
With the error:
`Unsupport cast from LocalDateTime to Long`

Then I use follow mehod for creating the table:
```java
 ConnectTableDescriptor desc = tableEnv.connect(pravega)
                .withFormat(new Json().failOnMissingField(true).deriveSchema())
                .withSchema(schema);

 final Map<String, String> propertiesMap = desc.toProperties();
 final TableSource<?> source = TableFactoryService.find(BatchTableSourceFactory.class, propertiesMap)
                .createBatchTableSource(propertiesMap);
 ConnectorCatalogTable<?, ?> connectorCatalogSourceTable = ConnectorCatalogTable.source(source, false);
 String tableSourcePath = tableEnv.getCurrentDatabase() + "." + "MyTableRow";

 tableEnv.getCatalog(tableEnv.getCurrentCatalog()).get().createTable(
                ObjectPath.fromString(tableSourcePath),
                connectorCatalogSourceTable, false);

```
with the following error:
```
Cannot add expression of different type to set:

set type is RecordType(VARCHAR(65536) CHARACTER SET "UTF-16LE" user, TIME ATTRIBUTE(ROWTIME) $f1, BIGINT NOT NULL cnt) NOT NULL

expression type is RecordType(VARCHAR(65536) CHARACTER SET "UTF-16LE" user, TIMESTAMP(3) NOT NULL $f1, BIGINT NOT NULL cnt) NOT NULL
```

The reason why I suspect the error has something to do with the way of creating tableSource is that `testTableSourceBatch` in testcase `testTableSource` run the same Sql successfully.And It uses the deprecated `FlinkPravegaJsonTableSource` api.

**Problem location**
`testTableSourceBatchDescriptor` in `io/pravega/connectors/flink/FlinkPravegaTableITCase.java`

**Suggestions for an improvement**
Come up with the correct way of creating TableSource in BatchTableEnvironment to avoid the error.

**Comment by crazyzhou** [comment]
Per my understanding, the `BatchTableEnvironment` is still using the old Flink SQL planner. Is there any Table API changes around that to cause this type interpreting issue?
There is a further question for this. Flink 1.10 changes the default planner to the Blink one. Instead of having different paths of streaming and batch for Table API in Flink planner, the Blink planner treats batch tables as a special case for streaming tables. Should we change our implementation for batch tables to adapt the Blink planner?
Invite some Flink experts @StephanEwen and @tzulitai for some advice and guidance. Thanks.

**Comment by crazyzhou** [comment]
Tracked in Flink issue: https://issues.apache.org/jira/browse/FLINK-16693

**Comment by crazyzhou** [comment]
After discussion, we agree to keep the legacy planner implementation now. The Blink planner change will be addressed in #356

**Comment by crazyzhou** [comment]
Close this issue as it should be fixed in #363 with the aligned behaviour of the new `DataType`.
